### DISCUSSION TITLE
How to access API externally written inside the Theia

### DISCUSSION THREAD

**Discussion by Gopikrishna-GitHB** [eclipse-theia/theia#12099] [Q&A]
Step 1: Create Theia application as a pod/Container in Kubernetes. (www.test.com) which is running in port 3000.
Step 2: Create a node app inside the Theia IDE 
Step 3: Try to access node App Externally outside the container through browser like Chrome.

@msujew @jfaltermeier Please share some idea if it possible. 
If not please let me know the technical reasons.
Thanks in Advance.

**Comment by msujew** [comment]
Hey @Gopikrishna-GitHB,

this sounds similar to what projects such as [Eclipse Che](https://github.com/eclipse/che) or [Gitpod](https://www.gitpod.io/) are already doing. So it's definitely possible, but you might not find someone who is able to guide you to building such a solution on your own. Theia is only a framework for building IDEs, and not much related to all the infrastructure that one might want to bundle with it. The infrastructure stuff is usually more related with Kubernetes and/or other orchestration services.
